runregistry-rest/api.py

```diff
@@ -0,0 +1,213 @@
+import os, io
+import flask
+from flask import Flask, request
+from flask_restful import Api, Resource
+from flask_caching import Cache
+from flask_sqlalchemy import SQLAlchemy
+from sqlalchemy import desc, func
+
+__all__ = ["app", "api", "db"]
+
+app = Flask(__name__)
+
+app.config["MAX_CONTENT_LENGTH"] = 32 * 1000 * 1000
+app.config["UPLOAD_EXTENSIONS"] = [".gz", ".tgz"]
+app.config["UPLOAD_PATH"] = "uploads"
+app.config["CACHE_TYPE"] = "simple"
+app.config["SQLALCHEMY_DATABASE_URI"] = os.environ.get(
+    "DATABASE_URI", "sqlite:////tmp/test.sqlite"
+)
+cache = Cache(app)
+db = SQLAlchemy(app)
+api = Api(app)
+
+from authentication import auth
+from database import RunRegistryMeta, RunRegistryConfig
+
+
+def cache_key():
+    args = flask.request.args
+    key = (
+        flask.request.path
+        + "?"
+        + urllib.urlencode(
+            [(k, v) for k in sorted(args) for v in sorted(args.getlist(k))]
+        )
+    )
+    return key
+
+
+def add_schema_as_element(rowres):
+    rowres.insert(0, queries.schema)
+
+
+# $ curl -u fooUsr:barPass -X GET np04-srv-021:5005/runregistry/getRunMeta/2
+@api.resource("/runregistry/getRunMeta/<int:runNum>")
+class getRunMeta(Resource):
+    """
+    Get XYZ
+    should return the RunMeta in format: XXXXXX
```

Comment:
Can you fill out the comments of this style?

Reply:
How do I upload a configuration to see what format it comes in?

Reply:
I think with a curl POST maybe?

Reply:
I'll see what I can do then